src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js

```diff
@@ -473,31 +473,130 @@ function prepareConstSumQuestions() {
+                 * are distributed among recipients
+                 */
+                $(`#constSumInputAlert-${qnNum}-${numResponses - 1}`).show();
+            } else {
+                /*
+                 * Display alert block for each recipient
+                 * since each recipient has its own set of options
+                 */
+                $(`[id^="constSumInputAlert-${qnNum}-"]`).show();
 
-                $(`#constSumInfo-${qnNum}-${numResponses - 1}`).show();
+                // Add further indentation to options for better layout if recipient name is visible
+                if ($(`.evalueeLabel-${qnNum}`).length) {
+                    $(`[id^="constSumSubmissionFormOptionFragment-${qnNum}"]`).addClass('padding-left-55px');
+                    $(`[id^="constSumSubmissionFormOptionFragment-${qnNum}"]`).addClass('margin-top-15px');
+                }
             }
         } else {
-            $(`[id^="constSumInfo-${qnNum}-"]`).hide();
+            $(`[id^="constSumInputAlert-${qnNum}-"]`).hide();
         }
     }
 }
 
-function updateConstSumMessageQn(qnNum) {
+/**
+ * Display instructions to be followed while distributing points
+ * Instructions can also be seen as the constraints that
+ * must be satisfied for arriving at a valid distribution
+ * @param qnNum - question number
+ */
+function updateConstSumQnInstructions(qnNum) {
     let numOptions = 0;
     let points = parseInt($(`#constSumPoints-${qnNum}`).val(), 10);
+
+    const numRecipients = parseInt($(`[name="questionresponsetotal-${qnNum}"]`).val(), 10);
+
+    // Boolean value stating the type of CONSTSUM question (among recipients or options)
+    const distributeToRecipients = $(`#constSumToRecipients-${qnNum}`).val() === 'true';
+    /*
+     * Boolean value stating whether total points to be distributed
+     * is equal to the value of variable `points` or `points`x no. of options
+     */
+    const pointsPerOption = $(`#constSumPointsPerOption-${qnNum}`).val() === 'true';
+    /*
+     * Boolean value which is true when all recipients/options can not be given
+     * the same amount of points.
+     * This variable is being used mainly for backward compatibility
+     */
+    const forceUnevenDistribution = $(`#constSumUnevenDistribution-${qnNum}`).val() === 'true';
+
+    /*
+     * For "among recipients" type of question, consider no. of options
+     * to be equal to the no. of recipients. Otherwise, extract it from
+     * the hidden input element `#constSumNumOption`
+     */
+    if (distributeToRecipients) {
+        numOptions = numRecipients;
+    } else {
+        numOptions = parseInt($(`#constSumNumOption-${qnNum}`).val(), 10);
+    }
+
+    if (pointsPerOption) {
+        points *= numOptions;
+    }
+
+    // Specifies the type of distribution constraint selected by question creator
+    const constSumDistributePointsOptionSelected = $(`#constSumDistributePointsOptions-${qnNum}`).val();
```

Comment:
Specifies if the constraint for uneven distribution of points is selected or something simpler